kratos/utilities/shifted_boundary_meshless_interface_utility.cpp

```diff
@@ -0,0 +1,1164 @@
+                }
+            }
+            aux_set = aux_extra_set;
+        }
+
+        // Sort the obtained nodes by id
+        const std::size_t n_cloud_nodes = aux_set.size();
+        rCloudNodes.resize(n_cloud_nodes);
+        std::size_t aux_i = 0;
+        for (auto it_set = aux_set.begin(); it_set != aux_set.end(); ++it_set) {
+            rCloudNodes(aux_i++) = *it_set;
+        }
+        std::sort(rCloudNodes.ptr_begin(), rCloudNodes.ptr_end(), [](NodeType::Pointer& pNode1, NodeType::Pointer rNode2){return (pNode1->Id() < rNode2->Id());});
+
+        // Fill the coordinates matrix
+        rCloudCoordinates.resize(n_cloud_nodes, 3);
+        IndexPartition<std::size_t>(n_cloud_nodes).for_each(array_1d<double,3>(), [&rCloudNodes, &rCloudCoordinates](std::size_t iNode, array_1d<double,3>& rAuxCoordTLS){
+            noalias(rAuxCoordTLS) = rCloudNodes[iNode].Coordinates();
+            rCloudCoordinates(iNode, 0) = rAuxCoordTLS[0];
+            rCloudCoordinates(iNode, 1) = rAuxCoordTLS[1];
+            rCloudCoordinates(iNode, 2) = rAuxCoordTLS[2];
+        });
+    }
+
+    double ShiftedBoundaryMeshlessInterfaceUtility::CalculateKernelRadius(
+        const Matrix& rCloudCoordinates,
+        const array_1d<double,3>& rOrigin)
+    {
+        const std::size_t n_nodes = rCloudCoordinates.size1();
+        const double squared_rad = IndexPartition<std::size_t>(n_nodes).for_each<MaxReduction<double>>([&](std::size_t I){
+            return std::pow(rCloudCoordinates(I,0) - rOrigin(0),2) + std::pow(rCloudCoordinates(I,1) - rOrigin(1),2) + std::pow(rCloudCoordinates(I,2) - rOrigin(2),2);
+        });
+        return std::sqrt(squared_rad);
+    }
+
+    std::size_t ShiftedBoundaryMeshlessInterfaceUtility::GetRequiredNumberOfPoints()
+    {
+        const std::size_t n_dim = mpModelPart->GetProcessInfo()[DOMAIN_SIZE];
+        switch (n_dim) {
+            case 2:
+                switch (mMLSExtensionOperatorOrder) {
+                    case 1:
+                        return 3;
+                    case 2:
+                        return 6;
+                    default:
+                        KRATOS_ERROR << "Wrong MLS extension operator order. Only linear (1) and quadratic (2) are supported.";
+                }
+            case 3:
+                switch (mMLSExtensionOperatorOrder) {
+                    case 1:
+                        return 4;
+                    case 2:
+                        return 10;
+                    default:
+                        KRATOS_ERROR << "Wrong MLS extension operator order. Only linear (1) and quadratic (2) are supported.";
+                }
+            default:
+                KRATOS_ERROR << "Wrong domain size.";
+        }
+    }
+
+    std::map<std::size_t, std::map<std::size_t, Vector>> ShiftedBoundaryMeshlessInterfaceUtility::SetSurrogateBoundaryNodalGradientWeights()
+    {
+        std::map<std::size_t, std::map<std::size_t, Vector>> sur_bd_nodes_map;

```

Comment:
maybe using unordered_map would be better?

Reply:
Agree. For the outer one it is possible. Not for the inner one.